examples/js/modules/paginator.js

```diff
@@ -0,0 +1,149 @@
+    } else {
+      const diff = totalImgCnt % cntPerPage;
+      endPage = cntPerPage;
+      let counter = 1;
+
+      // when images paginate into exactly 100 per page
+      if (diff === 0) {
+        pageCount = totalImgCnt / cntPerPage;
+        const i = pageCount;
+        while (pageCount >= 1) {
+          endPage = cntPerPage * counter;
+          initPage = (endPage - cntPerPage) + 1;
+          pageRange[i - pageCount] = `${initPage} - ${endPage} of ${totalImgCnt}`;
+
+          --pageCount;
+          ++counter;
+        };
+      } else {
+      // when the last page has less than 100 images
+        if (diff >= 1) {
+          pageCount = Math.trunc(totalImgCnt / cntPerPage);
+          let endPage = cntPerPage;
+          const i = pageCount;
+
+          while (pageCount >= 1) {
+            endPage = cntPerPage * counter;
+            initPage = (endPage - cntPerPage) + 1;
+            pageRange[i-pageCount] = `${initPage} - ${endPage} of ${totalImgCnt}`;
+            --pageCount;
+            ++counter;
+          };
+          --counter;
+          endPage = (cntPerPage * counter) + diff;
+          initPage = (endPage - diff) + 1;
+          pageRange[i - pageCount] = `${initPage} - ${endPage} of ${totalImgCnt}`;
+        };
+      };
+    };
+
+    return pageRange;
+  };
+
+  getTotalImageCount() {
+    return this.fetchedImages.filter(file => file.format === 'PNG' || file.format === 'JPEG').length;
+  };
+
+  processImgs(renderThumbnails, renderImages, count) {
+    thumbnailFunc = renderThumbnails;
+    imagesFunc = renderImages;
+    range.innerHTML = this.getRange()[this.currPage];
+
+    if (count > 100) {
+      if (this.imageThumbnails.length === this.fullResImages.length) {
+        renderThumbnails(this.imageThumbnails, this.url, this.fullResImages);
+      } else {
+        renderThumbnails(false, this.url, this.fullResImages);
+      }
+    } else {
+      renderImages(this.fullResImages, this.url);
+    }
+  };
+
+  clear() {
+    console.log('clear method ran');
+    this.nextBtn.removeEventListener('click', this.runNextPageMethod);
+    this.prevBtn.removeEventListener('click', this.runPrevPageMethod);
+  };
+
+  nextPage() {
+    const pages = this.paginate(this.fetchedImages.filter(file => file.format === 'PNG' || file.format === 'JPEG')).length;
+    if (this.currPage < pages) {
+      this.currPage = this.currPage + 1;
+    }
+
+    if (this.currPage === pages) {
+      this.currPage = 0;
+    }
+    imgContainer.textContent = '';
+
+    this.processImgs(thumbnailFunc, imagesFunc, this.count);
+  }
+
+  prevPage() {
+    const pages = this.paginate(this.fetchedImages.filter(file => file.format === 'PNG' || file.format === 'JPEG')).length;
+
+    if (this.currPage) {
+      this.currPage = this.currPage - 1;
+    } else {
+      this.currPage = pages - 1;
+    }
+
+    imgContainer.textContent = '';
+
+    this.processImgs(thumbnailFunc, imagesFunc, this.count);
+  }
+
+  imagesForPage(images) {
```

Comment:
Optional idea:

```suggestion
  getImagesForPage(images, page = this.currentPage) {
    return this.allPagesArray[page];
```

Reply:
Concerning this, I did attempt to implement this, but I ran into an issue that required me to run the loop which creates and displays the image in the sidebar.

What do you think? @jywarren